test/collection_vector_search_test.cpp

```diff
@@ -2229,4 +2230,174 @@ TEST_F(CollectionVectorTest, QueryByNotAutoEmbeddingVectorField) {
 
     ASSERT_EQ("Vector field `embedding` is not an auto-embedding field, do not use `query_by` with it, use `vector_query` instead.", search_res.error());
+}
+
+TEST_F(CollectionVectorTest, TestUnloadingModelsOnCollectionDelete) {
+    nlohmann::json actual_schema = R"({
+                        "name": "test",
+                        "fields": [
+                            {
+                                "name": "title",
+                                "type": "string"
+                            },
+                            {
+                            "name": "title_vec",
+                            "type": "float[]",
+                            "embed": {
+                                "from": [
+                                    "title"
+                                ],
+                                "model_config": {
+                                    "model_name": "ts/e5-small"
+                                }
+                            }
+                            }
+                        ]
+                        })"_json;
+    
+    TextEmbedderManager::set_model_dir("/tmp/typesense_test/models");
+
+    auto schema = actual_schema;
+    auto collection_create_op = collectionManager.create_collection(schema);
+    ASSERT_TRUE(collection_create_op.ok());
+
+    auto coll = collection_create_op.get();
+
+    auto text_embedders = TextEmbedderManager::get_instance()._get_text_embedders();
+
+    ASSERT_EQ(1, text_embedders.size());
+
+    auto delete_op = collectionManager.drop_collection("test", true);
+
+    ASSERT_TRUE(delete_op.ok());
+    text_embedders = TextEmbedderManager::get_instance()._get_text_embedders();
+    ASSERT_EQ(0, text_embedders.size());
+
+    // create another collection
+    schema = actual_schema;
+    collection_create_op = collectionManager.create_collection(schema);
+    ASSERT_TRUE(collection_create_op.ok());
+
+    coll = collection_create_op.get();
+
+    text_embedders = TextEmbedderManager::get_instance()._get_text_embedders();
+    ASSERT_EQ(1, text_embedders.size());
+
+    // create second collection
+    schema = actual_schema;
+    schema["name"] = "test2";
+    collection_create_op = collectionManager.create_collection(schema);
+    ASSERT_TRUE(collection_create_op.ok());
+
+    auto coll2 = collection_create_op.get();
+
+    text_embedders = TextEmbedderManager::get_instance()._get_text_embedders();
+
+    ASSERT_EQ(1, text_embedders.size());
+
+    delete_op = collectionManager.drop_collection("test", true);
+    ASSERT_TRUE(delete_op.ok());
+
+    text_embedders = TextEmbedderManager::get_instance()._get_text_embedders();
+    ASSERT_EQ(1, text_embedders.size());
+
+    delete_op = collectionManager.drop_collection("test2", true);
+    ASSERT_TRUE(delete_op.ok());
+
+    text_embedders = TextEmbedderManager::get_instance()._get_text_embedders();
+    ASSERT_EQ(0, text_embedders.size());
+}  
+
+TEST_F(CollectionVectorTest, TestUnloadingModelsOnDrop) {
```

Comment:
Can we add another test where the dropped collection has 2 embedding fields? In the past we have had issues where test covered only the single case.